packages/rum/src/domain/rumEventsCollection/view/trackViews.spec.ts

```diff
@@ -829,3 +829,88 @@ describe('rum view measures', () => {
     })
   })
 })
+
+describe('rum track custom timings', () => {
+  let setupBuilder: TestSetupBuilder
+  let handler: jasmine.Spy
+  let getViewEvent: (index: number) => View
+  let setCustomTiming: (name: string, inInitialView?: boolean, time?: number) => void
+
+  beforeEach(() => {
+    ;({ handler, getViewEvent } = spyOnViews())
+
+    setupBuilder = setup()
+      .withFakeLocation('/foo')
+      .beforeBuild(({ location, lifeCycle }) => {
+        lifeCycle.subscribe(LifeCycleEventType.VIEW_UPDATED, handler)
+        ;({ setCustomTiming } = trackViews(location, lifeCycle))
+      })
+  })
+
+  afterEach(() => {
+    setupBuilder.cleanup()
+  })
+
+  it('should add custom timing to current view', () => {
+    setupBuilder.build()
+    history.pushState({}, '', '/bar')
+    const currentViewId = getViewEvent(2).id
+    const now = performance.now()
+    setCustomTiming('foo', false, now)
+
+    const event = getViewEvent(3)
+    expect(event.id).toEqual(currentViewId)
+    expect(event.customTimings).toEqual({ foo: now - event.startTime })
+  })
+
+  it('should add custom timing to initial view', () => {
+    setupBuilder.build()
+    history.pushState({}, '', '/bar')
+    const initialViewId = getViewEvent(0).id
+    const now = performance.now()
+    setCustomTiming('foo', true, now)
+
+    const event = getViewEvent(3)
+    expect(event.id).toEqual(initialViewId)
+    expect(event.customTimings).toEqual({ foo: now - event.startTime })
+  })
+
+  it('should add multiple custom timings', () => {
+    setupBuilder.build()
+    const time1 = performance.now()
+    setCustomTiming('foo', false, time1)
+
+    const time2 = performance.now()
+    setCustomTiming('bar', false, time2)
+
+    const event = getViewEvent(2)
+    expect(event.customTimings).toEqual({
+      bar: time2 - event.startTime,
+      foo: time1 - event.startTime,
+    })
```

Comment:
You tests logic duplicate the implementation logic:

- time = performance.now() 
- customTime = time - startTime

It could be better to have control on start time and time and not reimplement the logic in the test to compute the output, something like:

```
setupBuilder.build()

clock.tick(20)
setCustomTiming('foo', false)

clock.tick(10)
setCustomTiming('bar', false)

expect(event.customTimings).toEqual({
      bar: 20,
      foo: 30,
})
```

Reply:
You are right, I updated the test.